smarts/core/vehicle_index.py

```diff
@@ -393,6 +388,11 @@ def build_agent_vehicle(
             boid,
         )
 
+        # Reserve space inside the traffic sim
+        sim._traffic_sim.reserve_traffic_location_for_vehicle(
```

Comment:
It is better to reserve the vehicle location in SUMO inside building the vehicle because this should be the case for any time an ego vehicle is added.

Reply:
@Gamenot You don't actually call this `build_agent_vehicle` from within `_switch_control_to_agent_recreate`. You have `Vehicle.build_agent_vehicle` instead of `self.build_agent_vehicle`. Is this intentional? It seems like we'd be skipping the traffic sim reservation step.

Reply:
This cross-talk between the vehicle and the traffic sim is breaking API boundaries. Traffic sim was private intentionally.

Reply:
I realize now that if the departure ordering works out I might be able to remove this part and have it only in the traffic sim.